[DOC] Update TraceQL metrics docs for release (#3418)
grafana/tempo: docs/sources/tempo/operations/traceql-metrics.md

Combined with the `by()` operator, this can be even more powerful!

```
{ resource.service.name = "foo" && status = error } | rate() by (span.http.route)
```

Now, we are still rating the erroring spans in the service "foo" but the metrics have been broken
down by http endpoint. This might let you determine that `/api/sad` had a higher rate of erroring
spans than `/api/happy` for example.

## Enable and use TraceQL metrics

You can use the TraceQL metrics in Grafana with any existing or new Tempo data source.
This capability is available in Grafana Cloud and Grafana (10.4 and newer).

![Metrics visualization in Grafana](/media/docs/tempo/metrics-explore-sample-2.4.png)

### Before you begin

To use the metrics generated from traces, you need to:

* Set the `local-blocks` processor to active in your `metrics-generator` configuration
* Configure a Tempo data source configured in Grafana or Grafana Cloud
* Access Grafana Cloud or Grafana 10.4

### Configure the `local-blocks` processor

Once the `local-blocks` processor is enabled in your `metrics-generator`
configuration, you can configure it using the following block to make sure
it records all spans for TraceQL metrics.

> 

Author: ```suggestion

Here is an example configuration: 
```

---
aliases: []
title: TraceQL metrics
menuTitle: TraceQL metrics
description: Learn about using TraceQL metrics.
weight: 550
keywords:
  - Prometheus
  - TraceQL
  - TraceQL metrics
---

# TraceQL metrics

{{< docs/experimental product="TraceQL metrics" >}}

Tempo 2.4 introduces the addition of metrics queries to the TraceQL language as an experimental feature.
Metric queries extend trace queries by applying a function to trace query results.
This powerful feature creates metrics from traces, much in the same way that LogQL metric queries create metrics from logs.
Initially, only `count_over_time` and `rate` are supported.

For example:
```
{ resource.service.name = "foo" && status = error } | rate()
```

> In this case we are rating the erroring spans in the service `foo`.

Reviewer: ```suggestion
In this case we are calculating the rate of the erroring spans coming from the service `foo`. Rate is a `spans/sec` quantity. 
```

Combined with the `by()` operator, this can be even more powerful!

```
{ resource.service.name = "foo" && status = error } | rate() by (span.http.route)
```

> Now, we are still rating the erroring spans in the service "foo" but the metrics have been broken

Author: ```suggestion
Now, we are still rating the erroring spans in the service `foo` but the metrics have been broken
```

> down by http endpoint. This might let you determine that `/api/sad` had a higher rate of erroring

Author: ```suggestion
down by HTTP endpoint. This might let you determine that `/api/sad` had a higher rate of erroring
```

> spans than `/api/happy` for example.

Author: ```suggestion
spans than `/api/happy`, for example.
```

## Enable and use TraceQL metrics

You can use the TraceQL metrics in Grafana with any existing or new Tempo data source.
This capability is available in Grafana Cloud and Grafana (10.4 and newer).

![Metrics visualization in Grafana](/media/docs/tempo/metrics-explore-sample-2.4.png)

### Before you begin

To use the metrics generated from traces, you need to:

* Set the `local-blocks` processor to active in your `metrics-generator` configuration
* Configure a Tempo data source configured in Grafana or Grafana Cloud
* Access Grafana Cloud or Grafana 10.4

### Configure the `local-blocks` processor

Once the `local-blocks` processor is enabled in your `metrics-generator`
configuration, you can configure it using the following block to make sure
it records all spans for TraceQL metrics.

Here is an example configuration: 
```yaml
 metrics_generator:
  processor:
    local_blocks:
      filter_server_spans: false
  storage:
    path: /tmp/tempo/generator/wal
  traces_storage:
    path: /tmp/tempo/generator/traces
```

> Refer to the [metrics-generator configuration]({{< relref "../configuration#metrics-generator">}}) documentation for more information.

Author: ```suggestion
Refer to the [metrics-generator configuration]({{< relref "../configuration#metrics-generator" >}}) documentation for more information.
```

### Evaluate query timeouts

> Because of the expensive nature, these queries can take a long time to run in different systems.

Reviewer 2: ```suggestion
Because of their expensive nature, these queries can take a long time to run in different systems.
```
